daemons/attrd/attrd_commands.c

```diff
@@ -252,10 +252,29 @@ void
 attrd_client_update(xmlNode *xml)
 {
     attribute_t *a = NULL;
-    char *host = crm_element_value_copy(xml, PCMK__XA_ATTR_NODE_NAME);
-    const char *attr = crm_element_value(xml, PCMK__XA_ATTR_NAME);
-    const char *value = crm_element_value(xml, PCMK__XA_ATTR_VALUE);
-    const char *regex = crm_element_value(xml, PCMK__XA_ATTR_PATTERN);
+    char *host;
+    const char *attr, *value, *regex;
+
+    /* Newer versions of the attribute manager support multiple updates in a
+     * single IPC message.  We handle that here by iterating over each and
+     * just calling attrd_client_update.  If the client didn't send a message
+     * with multiple updates (either because it's an older client version that
+     * doesn't know how to do that, or it's a newer client version that only
+     * needed to update one attribute) we skip this and proceed as normal.
+     */
+    if (xml_has_children(xml)) {
+        for (xmlNode *child = first_named_child(xml, XML_ATTR_OP); child != NULL;
+             child = crm_next_same_xml(child)) {
+            attrd_client_update(child);
+        }
+
+        return;
```

Comment:
We can broadcast the original op instead of splitting it up if all attrds support it. That would arrive at attrd_peer_update()

Reply:
Without modifications to attrd_peer_update, I don't see how this is going to work.  That function's going to bail immediately if there is no attribute, and there's no attribute in the top level of the multi-operation XML message.

Reply:
Yep, attrd_peer_update() is changeable :)

That's the point of tracking the protocol version -- we can know if all peers will support the new syntax or not.